jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c

```diff
@@ -49,31 +49,108 @@
  */
 
 /**
+ * Helper for stringifying numbers
+ *
+ * @return the length of the generated string representation
+ */
+static lit_utf8_size_t
+ecma_builtin_number_prototype_helper_to_string (lit_utf8_byte_t *digits, /**< number as string in decimal form */
+                                                lit_utf8_size_t num_digits, /**< length of the string representation */
+                                                int32_t exponent, /**< decimal exponent */
+                                                lit_utf8_byte_t *to_digits, /**< [out] buffer to write */
+                                                lit_utf8_size_t to_num_digits) /**< requested number of digits */
+{
+  lit_utf8_byte_t *p = to_digits;
```

Comment:
I've addressed all your comments but this. I've seen `p` used elsewhere, too, and couldn't come up with a better name myself.